update_factorio.py

```diff
@@ -138,11 +138,28 @@ def verbose_aware_exec(exec_args, verbose=False):
         raise
 
 
+def get_username_token(factoriopath):
+    with open(os.path.normpath(factoriopath + '/../../../player-data.json')) as data_file:  
```

Comment:
Headless servers won't have login credentials in `player-data.json`, and they're the ones most likely to use the updater (GUI versions have the internal updater or Steam).